compose-settings-storage-datastore/src/main/kotlin/com/alorma/compose/settings/storage/datastore/GenericDataStoreSettingValueState.kt

```diff
@@ -0,0 +1,48 @@
+package com.alorma.compose.settings.storage.datastore
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import com.alorma.compose.settings.storage.base.SettingValueState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+class GenericDataStoreSettingValueState<T>(
+  private val coroutineScope: CoroutineScope,
+  private val dataStore: DataStore<Preferences>,
+  private val dataStoreKey: Preferences.Key<T>,
+  private val defaultValue: T,
+) : SettingValueState<T> {
+
+  private var _value: T by mutableStateOf(runBlocking {
+    dataStore.data.map { preferences ->
+      preferences[dataStoreKey] ?: defaultValue
+    }.first()
+  })
+
+  override var value: T
+    get() = _value
+    set(value: T) {
+      _value = value
+      coroutineScope.launch {
```

Comment:
A job started by `coroutineScope.launch' can be incomplete by the time another job is started. I suggest saving the job to a variable and canceling it before starting a new job.

Something like:
```kotlin
_job?.cancel()
_job = coroutineScope.launch { ... }
```

Reply:
done